pkg/didcomm/protocol/introduce/states.go

```diff
@@ -151,22 +150,17 @@ func (s *arranging) ExecuteInbound(dis dispatcher.Outbound, m *metaData) (state,
 		recipient = m.Recipients[1]
 	}
 
-	return &noOp{}, dis.SendToDID(&Proposal{
+	return &noOp{}, messenger.Send(service.NewDIDCommMsgMap(Proposal{
 		Type:   ProposalMsgType,
 		ID:     uuid.New().String(),
 		To:     recipient.To,
 		Thread: &decorator.Thread{ID: m.ThreadID},
```

Comment:
I would expect the messenger to take care of threading, yes?

Reply:
It will, but it does not work yet. Also, the protocol will use ReplyTo instead of Send. I did it just to check that my custom function which converts struct to map works as expected.